src/Mod/Sketcher/Gui/SketcherSettings.ui

```diff
@@ -186,6 +186,70 @@ Requires to re-enter edit mode to take effect.</string>
         </property>
        </widget>
       </item>
+      <item>
+       <widget class="Gui::PrefCheckBox" name="checkBoxToggleSnap">
+        <property name="toolTip">
+         <string>Toggle snapping to objects. Snapping is overriding the mouse position to match the position of an object that is within the pick radius.</string>
+        </property>
+        <property name="text">
+         <string>Snap to objects</string>
+        </property>
+        <property name="checked">
+         <bool>true</bool>
+        </property>
+        <property name="prefEntry" stdset="0">
+         <cstring>SnapToObjects</cstring>
+        </property>
+        <property name="prefPath" stdset="0">
+         <cstring>Mod/Sketcher/General</cstring>
+        </property>
+       </widget>
+      </item>
+	  <item>
+       <layout class="QHBoxLayout" name="horizontalLayout">
+        <item>
+         <widget class="QLabel" name="label_1">
+          <property name="minimumSize">
+           <size>
+            <width>182</width>
+            <height>0</height>
+           </size>
+          </property>
+          <property name="text">
+           <string>Angular step for angle snapping (in degrees)</string>
+          </property>
+          <property name="buddy">
+           <cstring>spinBoxSnapAngle</cstring>
+          </property>
+         </widget>
+        </item>
+        <item>
+         <widget class="Gui::PrefDoubleSpinBox" name="spinBoxSnapAngle">
+          <property name="toolTip">
+           <string>Angular step that will be used for tools that have angle snapping (Slot, Line, Arc...) </string>
+          </property>
+          <property name="minimum">
+           <double>0.000000000000000</double>
+          </property>
+          <property name="maximum">
+           <double>360.000000000000000</double>
+          </property>
+          <property name="singleStep">
+           <double>1.000000000000000</double>
+          </property>
+          <property name="value">
```

Comment:
Maybe we shall check if value is a pure divider of 360, or we will have strange effects...

Reply:
I'm not sure what you mean. 
Do you mean in `updateSnapAngleParameters` we should make sure that 
` 0 <= snapAngle < 360` ?
By using fmod : 
`Client.snapAngle = fmod(hGrp->GetFloat("SnapAngle", 5.) * M_PI / 180, 2 * M_PI);`

Reply:
No, I was thinking to allow only values that are pure dividers of 360, because others will have strange effects.

Reply:
I used the same logic as arc tool, it feels more intuitive and less limiting. Accepting only pure deviders would be annoying for the user and probably feel weird "Why I can't set snapping to 25°?"

Reply:
I didn't know the arc tool in the Sketcher can be set to snap to specific angle values. :)

Reply:
I meant how the arc can turn 360 either counter clockwise or clockwise depending on which way you start.

Reply:
OK. Good to me. Now the problem maybe remaining is that your 'snapAtAngle' only use the vertical (0°) as reference direction, and not the actual direction where you start the arc -- I say this from reading the code, I didn't tried the PR yet --. I wonder if this is what user is expecting.

Reply:
Hmmm, not a strong answer, but as a user, if we have to end up with an uneven angle distribution, I'd expect the 0° reference to be the vertical (to up) axis.
Not strong opinion too, but when drawing an arc, I'd expect the angle to be counted from the start point to the end point. :confused:
Worth a discussion in the forum ?